apps/meteor/server/startup/migrations/v307.ts

```diff
@@ -0,0 +1,40 @@
+import { Apps } from '@rocket.chat/apps';
+import type { AppSignatureManager } from '@rocket.chat/apps-engine/server/managers/AppSignatureManager';
+import type { IAppStorageItem } from '@rocket.chat/apps-engine/server/storage';
+import { License } from '@rocket.chat/license';
+
+import type { AppRealStorage } from '../../../ee/server/apps/storage';
+import { addMigration } from '../../lib/migrations';
+
+addMigration({
+	version: 307,
+	name: "Mark all installed private apps as 'migrated'",
+	async up() {
+		const isEE = License.hasValidLicense();
```

Comment:
Shouldn't we flag them on EE workspaces as well, in case any of them switch to CE in the future?

Reply:
We don't need to because we're now handling this in the `onRemoveLicense` and `onInvalidateLicense` callbacks (check #33439). Private apps will always be flagged as migrated on downgrade starting on 7.0